css_coding_styleguide.md

```diff
@@ -0,0 +1,755 @@
+
+18F specifically does not recommend using Twitter/Bootstrap for production work because of one, the difficulty in adapting its opinionated styles to bespoke design work and two, its CSS style that places semantic layout instructions directly in HTML classes.
+
+
+## Format
+### Spacing
+- Where possible, limit CSS files’ width to 80 characters. See [notes](#format_notes) to see how to configure your text editor to 80 characters.
+	- There will be unavoidable exceptions to this rule, such as URLs, or gradient syntax. Don’t worry.
+- Use soft-tabs with a two space indent.
+- Put one space after : in property declarations.
+- Put spaces before { in rule declarations.
+- Put a blank line between each selector block.
+- To close a selector block, put an unindented closing curly brace on a separate line.
+- Each declaration should appear on its own line for more accurate error reporting.
+- Do not indent selectors.
+
+```scss
+// Good
+.rule {
+  margin: 3px;
+  text-align: center;
+}
+
+.another_rule {
+  margin: 3px;
+}
+
+// Bad
+.rule{
+    margin:3px;text-align:center;}
+```
+
+- Multiple selectors should each be on a single line, with no space after each comma, unless they selector is less than 5 chars.
+
+```scss
+// Good
+selector1,
+selector2,
+selector3 {
+}
+
+// Good
+h1, h2 {
+}
+
+// Bad 
+selector1, selector2 {
+}
+```
+
+### Property-Value Pairs
+- Put each pair on its own line.
+- Indent each pair one level.
+- End in a semicolon.
+
+```scss
+selector {
+  name: value;
+  name: value;
+}
+```
+
+
+- Do not use shorthand declarations unless you need to explicitly set all the available values.
+
+```scss
+// Bad
+margin: inherit 3em;
+
+// Good
+margin-bottom: 3em;
+margin-top: 3em;
+
+margin: 3em 4em 2em 1em;
+```
+
+- Single-quote URLs and string values.
+
+```scss
+  background-image: url('/images/kittens.jpg');
+  font-family: 'Helvetica', sans-serif;
+  font-family: 'Lucida Grande', 'Helvetica', sans-serif;
+```
+
+- Wrap top-level numeric calculations in parentheses.
+
+```scss
+// Good
+.component {
+  width: (100% / 3);
+}
+
+// Bad
+.component {
+  width: 100% / 3;
+}
+```
+
+- Avoid arbitrary numbers that are repeated, or linked, or dependent on other parts of the code, (aka “magic numbers”).
+
+```scss
+// Bad
+.component {
+  top: 0.327em;
+}
+
+// Better
+/**
+ * 1. Magic number. This value is the lowest I could find to align the top of
+ * `.foo` with its parent. Ideally, we should fix it properly.
+ */
+.component {
+  top: 0.327em;
+}
+
+// Good
+$align_top: 100%;
+.component {
+  top: $align_top;
+}
+```
+
+### Sorting
+* Follow ordering:
+	1. variables
+	2. @extend directives
+	3. @include directives
+	4. properties
+- Within properties, you may use alphabetical order or type order—just pick one and keep the whole project consistent.
+- Put a new line before nested selectors.
```

Comment:
We don't want a new line if you're nesting a selector (@extend or @include) directly after the first selector, right? That is, with no properties between them. If so, can we make this explicit?

```
// Good
.module {
  .module__ele {
    color: #fff932;
  }
}

// Bad
.module {

  .module__ele {
    color: #fff932;
  }
}
```